pkg/testing/define/requirements.go

```diff
@@ -0,0 +1,120 @@
+// you may not use this file except in compliance with the Elastic License.
+
+package define
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/elastic-agent/pkg/component"
+)
+
+const (
+	// Darwin is macOS platform
+	Darwin = component.Darwin
+	// Linux is Linux platform
+	Linux = component.Linux
+	// Windows is Windows platform
+	Windows = component.Windows
+)
+
+const (
+	// AMD64 is amd64 architecture
+	AMD64 = component.AMD64
+	// ARM64 is arm64 architecture
+	ARM64 = component.ARM64
+)
+
+// OS defines an operating system, architecture, version and distribution combination.
+type OS struct {
+	// Type is the operating system type (darwin, linux or windows).
+	//
+	// This is always required to be defined on the OS structure.
+	// If it is not defined the test runner will error.
+	Type string `json:"type"`
+	// Arch is the architecture type (amd64 or arm64).
+	//
+	// In the case that it's not provided the test will run on every
+	// architecture that is supported.
+	Arch string `json:"arch"`
+	// Version is a specific version of the OS type to run this test on
+	//
+	// When defined the test runs on this specific version only. When not
+	// defined the test is run on a selected version for this operating system.
+	Version string `json:"version"`
+	// Distro allows in the Linux case for a specific distribution to be
+	// selected for running on. Example would be "ubuntu".
+	Distro string `json:"distro"`
+}
+
+// Validate returns an error if not valid.
+func (o OS) Validate() error {
+	if o.Type == "" {
+		return errors.New("type must be defined")
+	}
+	if o.Type != Darwin && o.Type != Linux && o.Type != Windows {
+		return errors.New("type must be either darwin, linux, or windows")
+	}
+	if o.Arch != "" {
+		if o.Arch != AMD64 && o.Arch != ARM64 {
+			return errors.New("arch must be either amd64 or arm64")
+		}
+		if o.Type == Windows && o.Arch == ARM64 {
+			return errors.New("windows on arm64 not supported")
+		}
+	}
+	if o.Distro != "" && o.Type != Linux {
+		return errors.New("distro can only be set when type is linux")
+	}
+	return nil
+}
+
+// Cloud defines that a cloud instance is required.
+type Cloud struct {
+	// Version defines a specific cloud version to create for this test.
+	//
+	// In the case that no version is provided the same version being used for
+	// the current test execution is used.
+	Version string `json:"version"`
+}
+
+// Requirements defines the testing requirements for the test to run.
+type Requirements struct {
+	// OS defines the operating systems this test can run on. In the case
+	// multiple are provided the test is ran multiple times one time on each
+	// combination.
+	OS []OS `json:"os,omitempty"`
+
+	// Cloud defines that a cloud instance is required
+	Cloud *Cloud `json:"cloud,omitempty"`
+
+	// Local defines if this test can safely be performed on a local development machine.
+	// If not set then the test will not be performed when local only testing is performed.
```

Comment:
We should probably say something here about the fact that tests with this flag set to true could be run on a VM (e.g. in CI) so such tests shouldn't assume that they will always run on the same host as the test runner itself?

Reply:
Added more detail to the docstring.